src/t_set.c

```diff
@@ -1420,6 +1420,33 @@ void sinterCommand(client *c) {
     sinterGenericCommand(c, c->argv+1,  c->argc-1, NULL, 0, 0);
 }
 
+/* SMEMBERS key */
+void smembersCommand(client *c) {
+    setTypeIterator *si;
+    char *str;
+    size_t len;
+    int64_t intobj;
+    robj *setobj = lookupKeyRead(c->db, c->argv[1]);
+    if (checkType(c,setobj,OBJ_SET)) return;
+    if (!setobj) {
+        addReply(c, shared.emptyset[c->resp]);
+        return;
+    }
+
+    /* Prepare the response. */
+    addReplySetLen(c,setTypeSize(setobj));
```

Comment:
If the data is corrupted, setTypeSize(setobj) may not equal the number of replies in the following loop.
we can choose one of the following:
1. use addReplyDeferredLen().
2. add a assertion in the end.

fail CI: https://github.com/redis/redis/actions/runs/10692581371
@fcostaoliveira WDYT

Reply:
Fixed in https://github.com/redis/redis/pull/13515

Reply:
the second one is faster than the first.